tools/test/audio/comp_run.sh

```diff
@@ -5,14 +5,87 @@
 # stop on most errors
 set -e
 
-COMP=$1
-DIRECTION=$2
-BITS_IN=$3
-BITS_OUT=$4
-FS1=$5
-FS2=$6
-FN_IN=$7
-FN_OUT=$8
+usage ()
+{
+    echo "Usage:     $0 <options> <comp direction bits_in bits_out fs_in fs_out input output>"
+    echo "Example 1: $0 volume playback 16 16 48000 48000 input.raw output.raw"
+    echo "Example 2: $0 -t volume_config.sh"
+    echo
+    echo "Where volume_config.sh is"
+    echo
+    echo "COMP=volume"
+    echo "DIRECTION=playback"
+    echo "BITS_IN=16"
+    echo "BITS_OUT=16"
+    echo "CHANNELS_IN=2"
+    echo "CHANNELS_OUT=2"
+    echo "FS_IN=48000"
+    echo "FS_OUT=48000"
+    echo "FN_IN=input.raw"
+    echo "FN_OUT=output.raw"
+    echo "FN_TRACE=trace.txt"
+    echo "VALGRIND=yes"
+}
+
+# Defaults
+BITS_IN=16
+BITS_OUT=
+CHANNELS_IN=2
+CHANNELS_OUT=
+FS_IN=48000
+FS_OUT=
+VALGRIND=yes
+DEBUG=
+SOURCED_CONFIG=
+FN_TRACE=
+EXTRA_OPTS=
+
+while getopts ":ht:" opt; do
+    case "${opt}" in
+	h)
+	    usage
+	    exit
+	    ;;
+	t)
+	    source "${OPTARG}"
+	    SOURCED_CONFIG=yes
+	    ;;
+	*)
+	    usage
+	    exit 1
+	    ;;
+    esac
+done
+
+shift $((OPTIND -1))
+
+if [[ -z $SOURCED_CONFIG ]]; then
+    if [ $# -ne 8 ]; then
```

Comment:
"do || die" / assert style:
```suggestion
    [ -n "$SOURCED_CONFIG" ] || [ $# -eq 8 ] || {
        usage
        exit 1
        }
```

Reply:
This maybe works, I haven't tested with Matlab tests, but I don't think it improves readability:

```
    if ! "$SOURCED_CONFIG"; then
	[ $# -eq 8 ] || {
	    usage "$0"
	    exit 1
	}
	
	COMP=$1
	DIRECTION=$2
	BITS_IN=$3
	BITS_OUT=$4
	FS_IN=$5
	FS_OUT=$6
	FN_IN=$7
	FN_OUT=$8
    fi

```

Reply:
As above now, but I guess it's not what you were suggesting.